types/assertions.d.ts

```diff
@@ -138,6 +138,18 @@ export interface DeepEqualAssertion {
 	 */
 	<Actual, Expected extends Actual>(actual: Actual, expected: Expected, message?: string): actual is Expected;
 
+	/**
+	 * Assert that `actual` is [deeply equal](https://github.com/concordancejs/concordance#comparison-details) to
+	 * `expected`, returning a boolean indicating whether the assertion passed.
+	 */
+	<Actual extends Expected, Expected>(actual: Actual, expected: Expected, message?: string): expected is Actual;
+
+	/**
+	 * Assert that `actual` is [deeply equal](https://github.com/concordancejs/concordance#comparison-details) to
+	 * `expected`, returning a boolean indicating whether the assertion passed.
+	 */
+	<Actual, Expected>(actual: Actual, expected: Expected, message?: string): boolean;
```

Comment:
Why do we need this overload? Wouldn't the assertion always fail if neither extend one another?

Reply:
If part of `Expected` extended part of `Actual`, but another part `Expected` is a super type of `Actual`, then neither extends the other but both my have the same value.

For example, imagine that one of the properties on `Expected` is typed as readonly, while the corresponding property on `Actual` isn't. And then for another property the two share, the readonlyness is declared the other way around.